apps/vaporgui/PDatasetTransformWidget.cpp

```diff
@@ -12,7 +12,7 @@ using namespace VAPoR;
 PDatasetTransformWidget::PDatasetTransformWidget(VAPoR::ControlExec *ce)
 : PWidget("", _section = new VSectionGroup("Transform",
                                            {
-                                               _twDataset = new PStringDropdown(SelectedDatasetTag, {}, "Dataset"),
+                                               _twDataset = new PStringDropdown("transformWidgetDatasetTag", {}, "Dataset"),
```

Comment:
@shaomeng this fixes the problem and that is awesome! Nice job. But how about defining SelectedDatasetTag in a way that it will be properly initialized rather than hard-coding the string here? E.g. SelectedDatasetTag could be defined as a static member of PDatasetTransformWidget, or simply as an anonymous namespace variable within the PDatasetTransformWidget.cpp file? I think the former is what we do with all of the params classes.

Reply:
Well, I don't see other static data member usage across the family of PWidgets. Now I've found the problem and I'd rater defer the whatever proper fix to @StasJ .

Reply:
A note on static data members: Google C++ Guide simply bans such usage if the data member involves dynamic memory allocation. In our case, a static string would simply not be allowed.

Reply:
How about simply using constexpr to force compile time initialization then: https://en.cppreference.com/w/cpp/language/constexpr

Reply:
It seems only static data members can be declared as a constepr...

I was thinking that there are probably hundreds of PWidgets of all sorts, and there must be a mechanism that Stas is already using in those widgets.